src/lib/tree/Utils.sol

```diff
@@ -76,3 +76,31 @@ function _getSplitPoint(uint256 x) pure returns (uint256) {
     }
     return k;
 }
+
+/// @notice Returns the size of the subtree adjacent to `begin` that does
+/// not overlap `end`.
+/// @param begin Begin index, inclusive.
+/// @param end End index, exclusive.
+function _nextSubtreeSize(uint256 begin, uint256 end) pure returns (uint256) {
```

Comment:
I think you should include explicit function visibility for all added functions: https://docs.soliditylang.org/en/v0.8.22/contracts.html#function-visibility

Reply:
"free functions cannot have visibility"

Reply:
not so sure what to do. Should i keep the underscores? what about with the `_getSplitPoint`?

Reply:
Oh right. Nevermind then.